Remotion/ObjectBinding/Web.ClientScript/Scripts/BocAutoCompleteReferenceValue.ts

```diff
@@ -0,0 +1,356 @@
+      // The remaining browsers support ARIA 1.1 with NVDA.
+
+      var internetExplorerScreenReaderLabelID = baseID + '_InternetExplorerScreenReaderLabel';
+      var screenReaderLabel = $("<span hidden='hidden' />")
+        .attr("id", internetExplorerScreenReaderLabelID)
+        .html(resources.InternetExplorerScreenReaderLabelText)
+        .appendTo(combobox);
+
+      textbox.attr('role', combobox.attr('role'));
+      combobox.removeAttr('role');
+
+      textbox.attr('aria-haspopup', combobox.attr('aria-haspopup'));
+      combobox.removeAttr('aria-haspopup');
+
+      textbox.attr('aria-expanded', combobox.attr('aria-expanded'));
+      combobox.removeAttr('aria-expanded');
+
+      if (combobox.attr('aria-labelledby') !== undefined)
+      {
+        textbox.attr('aria-labelledby', combobox.attr('aria-labelledby') + ' ' + internetExplorerScreenReaderLabelID);
+        combobox.removeAttr('aria-labelledby');
+
+        if (combobox.attr('data-label-id-index') !== undefined)
+        {
+          textbox.attr('data-label-id-index', combobox.attr('data-label-id-index'));
+          combobox.removeAttr('data-label-id-index');
+        }
+      }
+      else
+      {
+        textbox.attr('aria-labelledby', internetExplorerScreenReaderLabelID);
+      }
+
+      if (combobox.attr('aria-describedby') !== undefined)
+      {
+        textbox.attr('aria-describedby', combobox.attr('aria-describedby'));
+        combobox.removeAttr('aria-describedby');
+      }
+
+      combobox = textbox;
+    }
+
+    const ClearError = () =>
+    {
+      if (textbox.hasClass('error'))
+      {
+        textbox.removeClass('error');
+        textbox.trigger('hideInformationPopUp');
+      }
+    }
+  
+    const SetError = (message: string) =>
+    {
+      textbox.addClass('error');
+      textbox.trigger('showInformationPopUp', { 'message': message });
+    }
+
+    const errorHandler = (error: Sys.Net.WebServiceError) =>
+    {
+      SetError(resources.LoadIconFailedErrorMessage);
+    };
+
+    textbox.autocomplete(this._controlServiceUrl, 'Search', 'SearchExact',
+      {
+        extraParams: searchContext,
+        isAutoPostBackEnabled: this._isAutoPostBackEnabled,
+        nullValue: this._nullValueString, // the hidden field value indicating that no value has been selected
+        searchStringValidationParams:
+        {
+          inputRegex: new RegExp(searchStringValidationInfo.ValidSearchStringRegex),
+          dropDownTriggerRegex: new RegExp(searchStringValidationInfo.ValidSearchStringForDropDownRegex),
+          dropDownTriggerRegexFailedMessage: searchStringValidationInfo.SearchStringForDropDownDoesNotMatchRegexMessage,
+          getDropDownSearchString: (searchString: string) =>
+          {
+            return searchStringValidationInfo.IgnoreSearchStringForDropDownUponValidInput ? this.GetDropDownSearchStringForValidInput(searchString) : searchString;
```

Comment:
Since a lambda is used to defined the function `this` from the outer scope is captured. The containing function is the constructor so `this` will be the instance passed to the constructor.